### DISCUSSION TITLE
magnetic field ploting

### DISCUSSION THREAD

**Discussion by yzz0198** [thliebig/openEMS-Project#121] [General]
hi, I was trying to calculate the maganetic field strength along an loop antenna's axis direction. I tried to plot the curve of the level of field verse the distance on its axis. it looks like I can not use the data from dump file directly. So can I have any suggestions or examples?
By the way. I am also curious about what it means that the data set 
![屏幕截图 2023-09-24 232735](https://github.com/thliebig/openEMS-Project/assets/55866581/86db3457-802d-4cb1-b05e-0fddb0b841ef)
been divided into 3 dimension array.

**Comment by LubomirJagos** [comment]
hi, here is code snippet to extract information from .h5 file, hope it's readable and self explain, structure of you H_field to get information from it is:
    >H_field.TD.values{time_index}(x_node_index, y_node_index, z_node_index, xyz_vector_index)
    
```matlab
#dump read
[portEtField portEtMesh] = ReadHDF5Dump([output_dir '/port_ET.h5']);

#
# Choose node using indexing inside E field box, there is mesh variable which hold distancies between nodes
#
for zCoord = 1:3
  figure;
  title(["E-field at Z layer " num2str(zCoord)]);

  for row = 1:7
    for col = 1:3
      portEx = [];
      portEy = [];
      portEz = [];

      nodeIndexX = row;
      nodeIndexY = col;
      nodeIndexZ = zCoord;

      for k = 1:length(portEtField.TD.values)
        portEx = [portEx portEtField.TD.values{k}(nodeIndexX, nodeIndexY, nodeIndexZ,1)];
        portEy = [portEy portEtField.TD.values{k}(nodeIndexX, nodeIndexY, nodeIndexZ,2)];
        portEz = [portEz portEtField.TD.values{k}(nodeIndexX, nodeIndexY, nodeIndexZ,3)];
      end

      portEx = portEx .* 1e-3;
      portEy = portEy .* 1e-3;
      portEz = portEz .* 1e-3;

      subplot(7,3,3*(row-1)+col);
      hold on;
      ylim([-1, 1]);
      plot(portEx);
      plot(portEy);
      plot(portEz);

      legend("X","Y","Z")

  ##    figure;
  ##
  ##    portETotal = abs(sqrt(portEx.^2 + portEy.^2 + portEz.^2))
  ##    title("E field total value");
  ##    plot(portETotal)

    end
  end
end
```
![image](https://github.com/thliebig/openEMS-Project/assets/18229825/3f5f392e-63e7-4b83-9e19-aada0a4d4eba)

**Reply by yzz0198** [reply]
so, what is the length of  the portEfield. TD. values, is the space steps or time steps?

**Reply by LubomirJagos** [reply]
plotEtField has this structure:
    - plotEtField.TD.values[] = array 1..18 for each timestep
        - plotEField.TD.values{time_step_index} = array (x_node_count_width, y_node_count_width, z_count_node_width, xyz_index)
        - plotEField.TD.names{time_step_index} = string, .vtk filename
        - plotEField.TD.time{time_step_index} = float, time in second
        
 Example:
     plotEtField.TD.values{5}(2,3,1,2) return value of Et field at 5th time step at node element X=2, Y=3, Z=1 and return Et_z scalar vector like -3.6381e-04 what is -0.36381 mV
     
![image](https://github.com/thliebig/openEMS-Project/assets/18229825/4ada2c5f-b6cf-4199-99d9-adad9f2b8fe2)

**Reply by yzz0198** [reply]
is FD work in same way?
